compute/disk-images/disk_image_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+func createDisk(ctx context.Context, projectId, zone, diskName, sourceImage string) error {
+	disksClient, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer disksClient.Close()
+	req := &computepb.InsertDiskRequest{
+		Project: projectId,
+		Zone:    zone,
+		DiskResource: &computepb.Disk{
+			Name:        proto.String(diskName),
+			SourceImage: proto.String(sourceImage),
+		},
+	}
+
+	op, err := disksClient.Insert(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func deleteDisk(ctx context.Context, projectId, zone, diskName string) error {
+	disksClient, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer disksClient.Close()
+	req := &computepb.DeleteDiskRequest{
+		Project: projectId,
+		Zone:    zone,
+		Disk:    diskName,
+	}
+
+	op, err := disksClient.Delete(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return op.Wait(ctx)
+}
+
+func TestComputeDiskImageSnippets(t *testing.T) {
+	ctx := context.Background()
+	var r *rand.Rand = rand.New(
+		rand.NewSource(time.Now().UnixNano()))
+	tc := testutil.SystemTest(t)
+	zone := "us-central1-a"
+	imageName := fmt.Sprintf("test-image-go-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	diskName := fmt.Sprintf("test-disk-go-%v-%v", time.Now().Format("01-02-2006"), r.Int())
+	sourceImage := "projects/debian-cloud/global/images/family/debian-11"
+
+	buf := &bytes.Buffer{}
+
+	err := createDisk(ctx, tc.ProjectID, zone, diskName, sourceImage)
```

Comment:
If you need to set up resources before running the test, create a `TestMain(m *testing.M)` method that creates the resources.

Reply:
No, I am *testing the disk creation.* Running this inside the test is fully intentional.